src/prop/cnf_stream.h

```diff
@@ -54,7 +59,122 @@ class CnfStream {
   typedef context::CDInsertHashMap<Node, SatLiteral, NodeHashFunction>
       NodeToLiteralMap;
 
+  /**
+   * Constructs a CnfStream that performs equisatisfiable CNF transformations
+   * and sends the generated clauses and to the given SAT solver. This does not
+   * take ownership of satSolver, registrar, or context.
+   *
+   * @param satSolver the sat solver to use.
+   * @param registrar the entity that takes care of preregistration of Nodes.
+   * @param context the context that the CNF should respect.
+   * @param outMgr Reference to the output manager of the smt engine. Assertions
+   * will not be dumped if outMgr == nullptr.
+   * @param fullLitToNodeMap maintain a full SAT-literal-to-Node mapping.
```

Comment:
Resource manager is missing from doc.

Reply:
Also, I assume you are adding the resource manager because it will be used?

Reply:
Will add. This was because it was used in the extension.